ouroboros-network/src/Ouroboros/Network/Server/Socket.hs

```diff
@@ -3,7 +3,10 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE BangPatterns #-}
 
-module Ouroboros.Byron.Proxy.Server.Socket
+-- `accept` is shadowed, but so what?
+{-# OPTIONS_GHC "-fno-warn-name-shadowing" #-}
+
+module Ouroboros.Network.Server.Socket
```

Comment:
So now that we're moving it here, it'll be more important to review it with @coot and @karknu, and document it more thoroughly. But I think we should be able to use it (perhaps with modifications) for both the proxy and the other use cases in `ouroboros-network`.

Reply:
The name usage in this module is rather confusing, finding where the real `accept` call on a real network `Socket` happens was a quest. Please call it something other than Socket, @semanticphilosopher  is good at coming up with names for new concept.